controllers/networkfence_controller.go

```diff
@@ -113,37 +118,66 @@ func (r *NetworkFenceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
+	nf := NetworkFenceInstance{
+		ctx:              ctx,
+		reconciler:       r,
+		logger:           logger,
+		instance:         nwFence,
+		controllerClient: client,
+	}
+
 	// check if the networkfence object is getting deleted, if yes, then send
 	// UnfenceCluster request to the driver.
 	// After the UnfenceCluster request is processed, the finalizer will be removed
 	// so that the networkfence object can be deleted gracefully.
-	if !nwFence.GetDeletionTimestamp().IsZero() {
+	if !nf.instance.GetDeletionTimestamp().IsZero() {
 		if util.ContainsInSlice(nwFence.GetFinalizers(), networkFenceFinalizer) {
 
-			err := r.unfenceClusterNetwork(ctx, nwFence, client, logger)
+			err := nf.deleteNetworkFenceResource()
 			if err != nil {
 				logger.Error(err, "failed to unfence cluster network")
 				return ctrl.Result{}, err
 			}
 		}
+
 		logger.Info("NetworkFence object is terminated, skipping reconciliation")
 		return ctrl.Result{}, nil
 	}
 
-	err = r.fenceClusterNetwork(ctx, nwFence, client, logger)
+	err = nf.processFencing()
 	if err != nil {
 		logger.Error(err, "failed to fence cluster network")
+		err = nf.updateStatus(csiaddonsv1alpha1.FencingOperationResultFailed, "fencing operation failed")
 		return ctrl.Result{}, err
```

Comment:
```suggestion
		UpdateErr = nf.updateStatus(csiaddonsv1alpha1.FencingOperationResultFailed, "fencing operation failed")
		if UpdateErr!=nil{
			logger.Error(UpdateErr, "failed to update status")
		}
		return ctrl.Result{}, err
```

The current format will not retrigger reconcile